### PR TITLE
adding privatekey auth param

### PR DESCRIPTION
**Pull request by iireland-ii**
Related Issues
fixes
https://github.com/deepset-ai/haystack-core-integrations/issues/1179

Proposed Changes:
Added PrivateKey and PrivateKeyPwd parameters to allow more secure connection methodologies.

How did you test it?
Tested locally with dev branch using live Snowflake instance and hatch.

### CODE REVIEW

File: integrations/snowflake/src/haystack_integrations/components/retrievers/snowflake/snowflake_table_retriever.py [hunk 1]
```diff
@@ -82,11 +84,16 @@ def __init__(
         :param db_schema: Name of the schema to use.
         :param warehouse: Name of the warehouse to use.
         :param login_timeout: Timeout in seconds for login. By default, 60 seconds.
+        :param private_key_file: Location of private key
```
Comment (davidsbatista) [inline integrations/snowflake/src/haystack_integrations/components/retrievers/snowflake/snowflake_table_retriever.py:87]: this docstring is very confusing - what's `key_file`? can you maybe rephrase it?

File: integrations/snowflake/src/haystack_integrations/components/retrievers/snowflake/snowflake_table_retriever.py [hunk 2]
```diff
@@ -69,6 +69,8 @@ def __init__(
         user: str,
         account: str,
         api_key: Secret = Secret.from_env_var("SNOWFLAKE_API_KEY"),  # noqa: B008
+        private_key_file: Optional[str] = None,
+        private_key_file_pwd: Optional[str] = None,
```
Comment (davidsbatista) [inline integrations/snowflake/src/haystack_integrations/components/retrievers/snowflake/snowflake_table_retriever.py:73]: this should be using the `Secret` - check above how it was done for `api_key`

**Comment by davidsbatista** [comment]
@iireland-ii thanks for your contribution! I've left a few comments

Can you also please test if an "old" serialized `SnowflakeTableRetriever` object can be deserialized with the code in this new version you are proposing?

File: integrations/snowflake/src/haystack_integrations/components/retrievers/snowflake/snowflake_table_retriever.py (hunk 1 above)
Comment (iireland-ii) [reply integrations/snowflake/src/haystack_integrations/components/retrievers/snowflake/snowflake_table_retriever.py:87]: I chose to match the snowflake connector params to keep it consistent with the snowflake documentation, see sample code here:
- https://docs.snowflake.com/developer-guide/python-connector/python-connector-connect#using-key-pair-authentication-and-key-pair-rotation

I've added some further comments to the docstring to include a note to the following link:
- https://docs.snowflake.com/en/user-guide/key-pair-auth

Comment (davidsbatista) [reply integrations/snowflake/src/haystack_integrations/components/retrievers/snowflake/snowflake_table_retriever.py:87]: I can't see your changes - also my comment was not about the params names themselves, rather the docstring explaining what they are used for.

**Comment by davidsbatista** [comment]
@iireland-ii do you still have time to work on this PR?

**Comment by iireland-ii** [comment]
hi david going to try make some time for it this week!

**Comment by Amnah199** [comment]
@iireland-ii If you have discontinued working on this, I'll close this PR as stale.

**Comment by Amnah199** [comment]
Closing as stale.
